File: Week2/Day-X/DayX/DetailViewController.m
//
//  DetailViewController.m
//  DayX
//
//  Created by Kaleo Kim on 5/4/15.
//  Copyright (c) 2015 DevMountain. All rights reserved.
//

#import "DetailViewController.h"
#import "EntryController.h"

@interface DetailViewController () <UITextFieldDelegate>
@property (weak, nonatomic) IBOutlet UITextField *textField;
@property (weak, nonatomic) IBOutlet UITextView *bodyTextView;

@end

@implementation DetailViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    [self updateWithEntry:self.entry];
}




- (IBAction)saveButtonTapped:(id)sender {
    
    if (self.entry) {
        self.entry.title = self.textField.text;
        self.entry.text = self.bodyTextView.text;
        self.entry.timestamp = [NSDate date];
//        [[EntryController sharedInstance] save];
        
        [[EntryController sharedInstance] updateEntry:self.entry];
    }
    else {
        [[EntryController sharedInstance] addEntryWithTitle:self.textField.text text:self.bodyTextView.text date:[NSDate date]];
//        self.entry = [[EntryController sharedInstance] createEntryWithTitle:self.textField.text bodyText:self.bodyTextView.text];
    }
}

-(void)updateWithEntry: (Entry *)entry {
    
    self.textField.text = entry.title;
    self.bodyTextView.text = entry.text;
    
}


- (IBAction)clearButtonTapped:(id)sender {
    self.textField.text = @"";
    self.bodyTextView.text = @"";
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (BOOL)textFieldShouldReturn:(UITextField *)textField
{
    [textField resignFirstResponder];
    return YES;
}
/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: Week7/SwiftDayX/SwiftDayX/DetailViewController.swift
//
//  DetailViewController.swift
//  SwiftDayX
//
//  Created by Kaleo Kim on 6/16/15.
//  Copyright (c) 2015 Kaleo Kim. All rights reserved.
//

import UIKit

class DetailViewController: UIViewController {

    @IBOutlet weak var titleTextField: UITextField!
    
    @IBOutlet weak var entryTextView: UITextView!
    
    var entry: Entry?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        if let entry = self.entry {
            self.updateWithEntry(entry)
        }
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func updateWithEntry(entry: Entry) {
        self.titleTextField.text = entry.title
        self.entryTextView.text = entry.text
    }
    
    @IBAction func saveButtonTapped(sender: AnyObject) {
        
        if let entry = self.entry {
            entry.title = self.titleTextField.text
            entry.text = self.entryTextView.text
            
            entryController.sharedInstance.save()
        } else {
            entryController.sharedInstance.createEntryWithDetails(titleTextField.text, text: entryTextView.text)
        }
        
        self.navigationController?.popViewControllerAnimated(true)
    }

    @IBAction func clearButtonTapped(sender: AnyObject) {
        titleTextField.text = ""
        entryTextView.text = ""
        
    }
    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
